### PR TITLE
Zamioculcas bin + project structure stuff

### PR DESCRIPTION
**Pull request by maclennann**
So this is a pretty giant PR. I'd be happy to break it up if need be.

Adding the `zamioculcas` commandline bin that can be used like `cat mypassword.txt | zamioculcas 2001 -d example.com -s scrt.example.com -k -t privatekey`. Plus some other options. Adding an example to the readme likely wouldn't have been a bad idea.

Using the built-in rspec rake task (+renaming files to match convention).

Hooked in travis and codeclimate to my fork as an example. Feel free to change this to run on not-my-fork or just get rid of it. But being able to automatically gem push on tagged commits is nice.

/cc @jdaviscooke

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-  Dir.chdir('test')
-  system("rspec zanzibar_spec.rb")
-end
+task :default => [:test]
+
+RSpec::Core::RakeTask.new(:test)
 
-task 'install_dependencies' do
-  system('bundle install')
+task :install_local do
```
Comment (jdaviscooke) [inline Rakefile]: Doesn't the gem_tasks 'rake install' do this already?

File: lib/zanzibar.rb [hunk 2]
```diff
@@ -10,9 +10,16 @@ module Zanzibar
   class Zanzibar
 
     ##
-    # @param args{:domain, :wsdl, :pwd, :globals{}}
+    # @param args{:domain, :wsdl, :pwd, :username, :globals{}}
 
     def initialize(args = {})
+
+      if args[:username]
```
Comment (jdaviscooke) [inline lib/zanzibar.rb]: Good call. This was on my list of things to add next.

**Comment by jdaviscooke** [comment]
I would say just add an example of the CL in use to the readme and then I'll merge this

File: spec/scrt.wsdl [hunk 3]
```diff
@@ -1,629 +1,629 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (maclennann) [inline spec/scrt.wsdl]: I don't know what Atom did to these files to make a simple rename show as a complete rewrite...

**Pull request merged**